src/transactions/feed/TransactionFeedV2.test.tsx

```diff
@@ -0,0 +1,217 @@
+    expect(tree.queryByTestId('NoActivity/loading')).toBeNull()
+    expect(tree.queryByTestId('TransactionList')).toBeNull()
+  })
+
+  it('renders correct status for a complete transaction', async () => {
+    mockFetch.mockResponse(typedResponse({ transactions: [mockTransaction()] }))
+    const tree = renderScreen()
+
+    await waitFor(() => tree.getByTestId('TransactionList'))
+    expect(tree.getByText('feedItemReceivedInfo, {"context":"noComment"}')).toBeTruthy()
+  })
+
+  it('renders correct status for a failed transaction', async () => {
+    mockFetch.mockResponse(
+      typedResponse({ transactions: [mockTransaction({ status: TransactionStatus.Failed })] })
+    )
+
+    const tree = renderScreen()
+
+    await waitFor(() => tree.getByTestId('TransactionList'))
+    expect(tree.getByText('feedItemFailedTransaction')).toBeTruthy()
+  })
+
+  it('tries to fetch 20 transactions, unless the end is reached', async () => {
+    mockFetch
+      .mockResponseOnce(
+        typedResponse({
+          transactions: [mockTransaction({ transactionHash: '0x01', timestamp: 10 })],
+        })
+      )
+      .mockResponseOnce(
+        typedResponse({
+          transactions: [mockTransaction({ transactionHash: '0x02', timestamp: 20 })],
+        })
+      )
+      .mockResponseOnce(typedResponse({ transactions: [] }))
+
+    const { store, ...tree } = renderScreen()
+
+    await waitFor(() => tree.getByTestId('TransactionList'))
+
+    fireEvent(tree.getByTestId('TransactionList'), 'onEndReached')
+    await waitFor(() => expect(mockFetch).toBeCalled())
+    await waitFor(() => expect(tree.getByTestId('TransactionList/loading')).toBeVisible())
+    await waitFor(() => expect(tree.queryByTestId('TransactionList/loading')).toBeFalsy())
+
+    fireEvent(tree.getByTestId('TransactionList'), 'onEndReached')
+    await waitFor(() => expect(mockFetch).toBeCalled())
+    await waitFor(() => expect(tree.getByTestId('TransactionList/loading')).toBeVisible())
+    await waitFor(() => expect(tree.queryByTestId('TransactionList/loading')).toBeFalsy())
+
+    expect(mockFetch).toHaveBeenCalledTimes(3)
+    expect(getNumTransactionItems(tree.getByTestId('TransactionList'))).toBe(2)
+  })
+
+  it('tries to fetch 10 transactions, and stores empty pages', async () => {
+    mockFetch
+      .mockResponseOnce(typedResponse({ transactions: [mockTransaction()] }))
+      .mockResponseOnce(typedResponse({ transactions: [] }))
+
+    const { store, ...tree } = renderScreen()
+
+    await store.dispatch(
+      transactionFeedV2Api.endpoints.transactionFeedV2.initiate({ address: '0x00', endCursor: 0 })
+    )
+    await store.dispatch(
+      transactionFeedV2Api.endpoints.transactionFeedV2.initiate({ address: '0x00', endCursor: 123 })
+    )
+
+    await waitFor(() => tree.getByTestId('TransactionList'))
+
+    await waitFor(() => expect(mockFetch).toHaveBeenCalledTimes(2))
+    expect(getNumTransactionItems(tree.getByTestId('TransactionList'))).toBe(1)
+  })
```

Comment:
Do we really need this one?

Reply:
@jeanregisser I think you're right that we can skip this. In the next PR I've added a test that checks for correct lazy-load of a few pages of data so that should cover the same thing this test was covering.